multi-agent/lib/mas/resources/service.py

```
@@ -78,9 +78,74 @@ def update(self, rid: str, *, config: dict, name: str = None) -> Resource:
+        resource = self._store.get(rid)
+        bp_ids, res_ids = self._store.check_usage(rid)
+
+        if not bp_ids and not res_ids:
+            return {"in_use": False}
+
+        bp_details = []
+        for bp_id in bp_ids:
+            try:
+                bp_doc = self._store.get_blueprint_summary(bp_id)
+                bp_details.append(bp_doc)
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)